docs/try/tutorials/geomodel/geomodel-basics.md

```diff
@@ -1,47 +1,153 @@
+ * Print statistics on the imported geomodel.
+ * Mesh it with TetGen.
+ * Print statistics on the imported geomodel.
+ * Save it to the GINGMesh file format (.gm) `ModelA1.gm`.
+ * Open and analyse the .gm file [see description](/features/file_formats).
  
-The `RINGMesh::GeoModelBuilder` implements functions to build and modify the elements of a `RINGMesh::GeoModel`.
-Any new builder should derive from this class.
-Three main builders are implemented:
-
- * RINGMesh::GeoModelBuilderGM to load RINGMesh geological model format
-  (boundary representation or 3D meshed models).
- * RINGMesh::GeoModelBuilderGocad to load either:
-    * A SKUA-GOCAD structural model saved in a .ml (RINGMesh::GeoModelBuilderML) file.
-    * A volumetric model saved in a .so (RINGMesh::GeoModelBuilderTSolid) file.
- * RINGMesh::GeoModelBuilderSurfaceMesh and RINGMesh::GeoModelBuilderMesh to build a
-RINGMesh::GeoModel from a RINGMesh::Mesh.
-
-###GeoModelMesh: How to use the global iteration?
-
-    // Iterates on the RINGMesh::GeoModelMesh (gmm) vertices without redundancy
-    for( index_t v = 0; v < gmm.vertices.nb_vertices(); v++ ) {
-        const vec3& point = gmm.vertices.vertex( v ) ;
+### Manipulate GeoModelEntities
+
+There are two kinds of GeoModelEntities: GeoModelMeshEntities such as Corners, Lines, Surfaces and Regions
+and GeoModelGeologicalEntities such as Contact, Interface, Layer (but you also can add yours) [see GeoModel](/features/geomodel).
+
+The following consists in two code samples about each kind of GeoModelEntities usage.
+
+#### GeoModelMeshEntities
+
+    // Iterates on the Surfaces of a GeoModel and looks for boundaries and incident_entities
+    for( const auto& surface : geomodel.surfaces() ) 
+    {
+        for( auto boundary_id : range( surface.nb_boundaries() ) // Iterates on boundary lines 
+        {
+            const auto& line = surface.boundary( boundary_id );
+            ...
+        }
+        for( auto incident_id : range( surface.nb_incident_entities() ) // Iterates on incident regions (max. 2 incident regions) 
+        {
+            const auto& region = surface.incident_entity( incident_id );
+            ...
+        }
+    }
+
+#### GeoModelGeologicalEntities
+
+    // Iterates on the Interfaces of a GeoModel and looks for children (being GeoModelMeshEntities)
+    for( const auto& interface : geomodel.geol_entities( Interface::type_name_static() ) ) 
+    {
+        for( auto child_id : range( interface.nb_children() ) // Iterates on boundary lines 
+        {
+            const auto& surface = interface.child( child_id );
+            ...
+        }
+    }
+    
+    // Iterates on the Surfaces of a GeoModel and looks for them parent of type Interface
+    for( const auto& surface : geomodel.surfaces() ) 
+    {
+        if( surface.has_parent( Interface::type_name_static() ) )
+        {
+            const auto& interface = surface.parent( Interface::type_name_static() );
```

Comment:
This line is ok as example but will not compile on Windows thanks to the interface keyword ;)